### ISSUE TITLE
[Subtask] Support PaimonCatalog implementation to manage Paimon table operations

### ISSUE THREAD

**Issue by SteNicholas** [datastrato/gravitino#2738]
### Describe the subtask

Support `PaimonCatalog` implementation to provide Gravitino catalog for Paimon table operation management.

### Parent issue

#1129

**Comment by SteNicholas** [comment]
@FANNG1, could you assign this ticket to me?

**Comment by KarldBai** [comment]
Do we have a vague ETA for these? Maybe not all done but generally ready to go or play around with? Thank you

**Comment by FANNG1** [comment]
yes, Gravitino had supported the basic features of Paimon and will be released in 0.6.0,  please refer to  https://github.com/apache/gravitino/blob/main/docs/lakehouse-paimon-catalog.md,

**Comment by KarldBai** [comment]
Got it! Thanks
